backend/utils.py

```diff
@@ -168,7 +168,10 @@ def format_pf_non_streaming_response(
                         {
                             "role": "assistant",
                             "content": chatCompletion[response_field_name],
-                        }
+                        },
+                        {
+                            "role": "tool",
+                            "content": chatCompletion[citations_field_name]
```

Comment:
Missing closing brace on the message object.  

Also, I'm wondering if the citations field is guaranteed to be present in all cases?  Just wondering if we should consider checking for the citations field before adding it to messages by default.

Reply:
Citations is always present as long as endpoint response code is 200, it will be an empty array when no relevant documents were found